### ISSUE TITLE
Page-flip integration in ngx-extended-pdf-viewer

### ISSUE THREAD

**Issue by stephanrauh** [RaffaeleMorganti/pdf-viewer#6]
Hi @RaffaeleMorganti ,

first of all, thanks for implementing this! The viewer looks nice.

I've like to integrate it with my own Angular PDF viewer. It's based on Mozilla's pdf.js, so in theory it should be possible. Would you like to give me a few hints how to do it? Or (just hoping :)) would you like to contribute it yourself?

https://github.com/stephanrauh/ngx-extended-pdf-viewer/issues/161

https://www.npmjs.com/package/ngx-extended-pdf-viewer

Best regards,
Stephan

**Comment by stephanrauh** [comment]
@RaffaeleMorganti Ping? Are you still supporting this project?

**Comment by yashgautami** [comment]
> Hi @RaffaeleMorganti ,
> 
> first of all, thanks for implementing this! The viewer looks nice.
> 
> I've like to integrate it with my own Angular PDF viewer. It's based on Mozilla's pdf.js, so in theory it should be possible. Would you like to give me a few hints how to do it? Or (just hoping :)) would you like to contribute it yourself?
> 
> [stephanrauh/ngx-extended-pdf-viewer#161](https://github.com/stephanrauh/ngx-extended-pdf-viewer/issues/161)
> 
> https://www.npmjs.com/package/ngx-extended-pdf-viewer
> 
> Best regards,
> Stephan

Hi @stephanrauh ,

You found any solution for this?

**Comment by stephanrauh** [comment]
@yashgautami Yes and no. I've implemented a prototype that almost worked - but each time I had a look at it, I found another bug. After several week, I've put the project aside.

In the meantime, I've started to maintain my own fork of pdf.js. So I can modify it much more boldly. However, I didn't manage to find a really nice page flip effect algorithm. Using turn.js is a no-go because it uses jQuery and - more importantly - because it has been published under a restrictive license.

**Comment by stephanrauh** [comment]
Hm, looking at turn.js again... maybe the license is almost compatible to the license of ngx-extended-pdf-viewer. But still, I'd prefer a solution without jQuery.
